go/vt/discovery/healthcheck.go

```diff
@@ -427,44 +427,48 @@ func (hc *HealthCheckImpl) updateHealth(th *TabletHealth, shr *query.StreamHealt
 	// add it to the map by target
 	hc.healthData[targetKey][tabletAlias] = th
 
-	if isMasterUpdate {
-		if len(hc.healthy[targetKey]) == 0 {
-			hc.healthy[targetKey] = append(hc.healthy[targetKey], th)
-		} else {
-			// We already have one up server, see if we
-			// need to replace it.
-			if shr.TabletExternallyReparentedTimestamp < hc.healthy[targetKey][0].MasterTermStartTime {
-				log.Warningf("not marking healthy master %s as Up for %s because its MasterTermStartTime is smaller than the highest known timestamp from previous MASTERs %s: %d < %d ",
-					topoproto.TabletAliasString(shr.TabletAlias),
-					topoproto.KeyspaceShardString(shr.Target.Keyspace, shr.Target.Shard),
-					topoproto.TabletAliasString(hc.healthy[targetKey][0].Tablet.Alias),
-					shr.TabletExternallyReparentedTimestamp,
-					hc.healthy[targetKey][0].MasterTermStartTime)
+	if th.Target.TabletType == topodata.TabletType_MASTER {
```

Comment:
I would presume that `isPrimaryUp` would be true only if the tablet type was MASTER. Is there a case where this is not the case?

Reply:
That is a good point. I suppose this is more of a consistency check - to make sure that a caller doesn't pass in `isPrimaryUp` true for a non-master tablet type.